### PR TITLE
Mds for Sample, Table, Tensor, MiniBatch, Dataset

### LINKED ISSUE #1202: Missing md docs for Sample, Table, Tensor, MiniBatch, Dataset
Please add them in `docs/docs/APIdocs/Data`

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?
Mds for Sample, Table, Tensor, MiniBatch, Dataset

## Related links or issues (optional)
fixed #1202

### CODE REVIEW

File: docs/docs/APIdocs/Data/Tensor.md [hunk 1]
```diff
@@ -1,16 +1,71 @@
...
+res: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+0.16666667	0.4	0.75
+1.3333334	2.5	6.0
+[com.intel.analytics.bigdl.tensor.DenseTensor of size 2x3]
+```
+For more API, please visit [bigdl scala doc](https://bigdl-project.github.io/latest/#APIdocs/scaladoc/)
+
+**Python API:**
+
+Python API use numpy instead of `Tensor`, see [numpy](http://www.numpy.org/) for details.
```
Comment (shane-huang) [inline docs/docs/APIdocs/Data/Tensor.md]: If we don't have Python API, we don't need to add it here. Just remove line 68-70. 

We will describe the corresponding relationships in Getting Started, Programming Gude and examples.

Comment (shane-huang) [inline docs/docs/APIdocs/Data/Tensor.md]: Others LGTM

File: docs/docs/APIdocs/Data/Tensor.md [hunk 2]
```diff
@@ -1,16 +1,67 @@
...
- tensor: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+ scala> val tensor = Tensor(2, 3)
+ tensor: com.intel.analytics.bigdl.tensor.Tensor =
  0.0     0.0     0.0
  0.0     0.0     0.0
  [com.intel.analytics.bigdl.tensor.DenseTensor of size 2x3]
 ```
+Tensor can be created with existing data.
+```scala
+scala> val a = Tensor(Array(1f, 2, 3, 4, 5, 6), Array(2, 3))
```
Comment (yiheng) [inline docs/docs/APIdocs/Data/Tensor.md]: change the example by using the Table

File: docs/docs/APIdocs/Data/Dataset.md [hunk 3]
```diff
@@ -1 +1,56 @@
-## Dataset
+## DataSet
+`DataSet` is a set of data which is used in the model optimization process. You can use `DataSet.array()` and `DataSet.rdd()` function to create an Dataset.
```
Comment (yiheng) [inline docs/docs/APIdocs/Data/Dataset.md]: Mentioned that user should use rdd API of optimizer.

File: docs/docs/APIdocs/Data/MiniBatch.md [hunk 4]
```diff
@@ -1 +1,52 @@
...
+`MiniBatch` can be created by `MiniBatch(nInputs: Int, nOutputs: Int)`, `nInputs` means number of inputs, `nOutputs` means number of outputs. And you can use `set(samples: Seq[Sample[T])` to fill the content in this MiniBatch.
+
+**Scala example:**
+```scala
+import com.intel.analytics.bigdl.dataset.Sample
+import com.intel.analytics.bigdl.dataset.MiniBatch
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.numeric.NumericFloat
+
+val samples  = Array.tabulate(5)(i => Sample(Tensor(1, 3, 3).fill(i), i + 1f))
```
Comment (yiheng) [inline docs/docs/APIdocs/Data/MiniBatch.md]: Mentioned that user should use rdd API of optimizer.

File: docs/docs/APIdocs/Data/Tensor.md [hunk 5]
```diff
@@ -1,16 +1,75 @@
...
- tensor: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+ scala> val tensor = Tensor(2, 3)
+ tensor: com.intel.analytics.bigdl.tensor.Tensor =
  0.0     0.0     0.0
  0.0     0.0     0.0
  [com.intel.analytics.bigdl.tensor.DenseTensor of size 2x3]
 ```
+Tensor can be created with existing data.
+```scala
+scala> val a = Tensor(T(1f, 2f, 3f, 4f, 5f, 6f)).resize(2, 3)
```
Comment (yiheng) [inline docs/docs/APIdocs/Data/Tensor.md]: Tensor(
T(
    T(1f, 2f, 3f),
    T(4f, 5f, 6f)
))

File: docs/docs/APIdocs/Data/Tensor.md [hunk 6]
```diff
@@ -1,16 +1,75 @@
...
+a: com.intel.analytics.bigdl.tensor.Tensor[Float] =
+1.0	2.0	3.0
+4.0	5.0	6.0
+[com.intel.analytics.bigdl.tensor.DenseTensor$mcF$sp of size 2x3]
+
+scala> val b = Tensor(T(6f, 5f, 4f, 3f, 2f, 1f)).resize(2, 3)
```
Comment (yiheng) [inline docs/docs/APIdocs/Data/Tensor.md]: same

**Pull request merged**